### PR TITLE
V2 addon format and switch to pnpm, change localization and styles setup

### PR DESCRIPTION
**Pull request by AmauryD**
Related issue: #1910
@RobbieTheWagner I began to work on this, but this will be a breaking change:

> treeFor, treeForAddon, treeForAddonTemplates, treeForAddonTestSupport, treeForApp, treeForPublic, treeForStyles, treeForTemplates, treeForTestSupport, treeForVendor: Dropped. Dynamically generating broccoli trees at app build time is no longer supported. Your trees are built at publication time. If you need to produce different output at build time based on dynamic configuration, see Build-time Conditionals.

The theme and language part of flatpickr will be impacted.

### CODE REVIEW

**Comment by RobbieTheWagner** [comment]
@AmauryD yeah, we will need to tell people to import the themes and languages directly, which I think is fine.

**Comment by RobbieTheWagner** [comment]
Maybe flatpickr should be be a peerDep as well, so people ship all their own flatpickr stuff, but we provide the components on top.

**Comment by AmauryD** [comment]
> @AmauryD yeah, we will need to tell people to import the themes and languages directly, which I think is fine.

> Maybe flatpickr should be be a peerDep as well, so people ship all their own flatpickr stuff, but we provide the components on top.

I agree, let's do this !

**Comment by AmauryD** [comment]
@RobbieTheWagner I'm having a hard time figuring how to tell the user to import themes/default locales correctly. Does this seems ok to you ? https://github.com/shipshapecode/ember-flatpickr/blob/8ea4e52ba814e3ace92eff85077f5c0d626f9c16/test-app/app/templates/docs/usage.md
I only made the theme import work by using a js import. Didn't seems to work from the css stylesheet.

**Comment by AmauryD** [comment]
I'm currently testing this PR in one of my company's project. Seems to be working fine. Just some documentation to finish and it's ready.

**Comment by RobbieTheWagner** [comment]
@AmauryD what's the status on this?

**Comment by AmauryD** [comment]
@RobbieTheWagner should be ready for review !

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,67 +23,20 @@ jobs:
...
-          node-version: 16.x
-          cache: yarn
-      - name: Install Dependencies
-        run: yarn install --frozen-lockfile
-      - name: Run Tests
-        run: ./node_modules/.bin/ember try:one ${{ matrix.try-scenario }}
-
+        run: pnpm test
   automerge:
-    needs: [test, floating, try-scenarios]
```
Comment (RobbieTheWagner) [inline .github/workflows/ci.yml:81]: @AmauryD can we add back the ember-try scenarios please?

File: .gitignore [hunk 2]
```diff
@@ -1,34 +1,11 @@
...
-/dist/
-/tmp/
-
 # dependencies
-/bower_components/
-/node_modules/
+node_modules/
 
 # misc
-/.env*
```
Comment (RobbieTheWagner) [inline .gitignore:12]: I think pretty much everything should stay as it was in this file and the old contents should be copied to the new ember-flatpickr folder, I think, to ensure the right things are ignored. Especially stuff like ember-try, which we want to keep.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AmauryD) [reply .github/workflows/ci.yml:81]: is it necessary ? It's already tested by the test command. https://github.com/AmauryD/ember-flatpickr/blob/main/test-app/config/ember-try.js

https://github.com/AmauryD/ember-flatpickr/blob/6afd587d5806042b123cfe44531b398cede8485c/test-app/package.json#L35

Comment (RobbieTheWagner) [reply .github/workflows/ci.yml:81]: The ember-try setup should remain as close to how it was before as possible for ease of upgrading.

Comment (RobbieTheWagner) [reply .github/workflows/ci.yml:81]: Also, I don't know if it runs in CI at all now. We want each try scenario to be a separate run so we can see which ones pass and fail.

Comment (AmauryD) [reply .github/workflows/ci.yml:81]: I understand. I'll fix this after work 😄

Comment (RobbieTheWagner) [reply .github/workflows/ci.yml:81]: Thank you for all your hard work on this!

**Comment by AmauryD** [comment]
Downgraded test helpers from 3.x to 2.x to pass compatibility of ember-lts-4.4

File: patches/ember-cli-addon-docs@6.0.2.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
+diff --git a/.vscode/settings.json b/.vscode/settings.json
+new file mode 100644
+index 0000000000000000000000000000000000000000..f6e1d59217c5a9a7b520c5b78336c519ca2360f1
+--- /dev/null
++++ b/.vscode/settings.json
```
Comment (RobbieTheWagner) [inline patches/ember-cli-addon-docs@6.0.2.patch:5]: @AmauryD what is this for? Do we need this file added to ember-cli-addon-docs?

Comment (AmauryD) [reply patches/ember-cli-addon-docs@6.0.2.patch:5]: There is problem with recent versions of ember-data with ember-cli-addon-docs as mentionned here https://github.com/ember-learn/ember-cli-addon-docs/issues/1507. I made a quick patch until it's fixed.

File: test-app/.prettierignore [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# unconventional js
+/blueprints/*/files/
+
+# compiled output
+/dist/
+
+# misc
+/coverage/
```
Comment (RobbieTheWagner) [inline test-app/.prettierignore:8]: Does this need to ignore node_modules or does it by default?

File: test-app/.stylelintignore [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# unconventional files
+/blueprints/*/files/
+
```
Comment (RobbieTheWagner) [inline test-app/.stylelintignore:3]: Does this need to ignore node_modules or does it by default?

Comment (AmauryD) [reply test-app/.stylelintignore:3]: The default configuration seems to be ignoring this by default, but I couldn't find any reference to it in the documentation. In the base `ember new` template, there's no line for node_modules in the .styleignore file.

File: test-app/.prettierignore (hunk 4 above)
Comment (AmauryD) [reply test-app/.prettierignore:8]: it does by default !

File: patches/ember-cli-addon-docs@6.0.2.patch (hunk 3 above)
Comment (AmauryD) [reply patches/ember-cli-addon-docs@6.0.2.patch]: I've just updated to ember-cli-addon-docs v7. No more patch needed ! 🚀

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AmauryD) [reply .github/workflows/ci.yml:81]: The scenarios are back !

File: .gitignore (hunk 2 above)
Comment (AmauryD) [reply .gitignore:12]: Just to be sure, you prefer to keep the original content of this file at the project root ?

**Comment by AmauryD** [comment]
Merry PR Chistmas 🎅 ⛄

File: test-app/app/templates/flatpickr-test.hbs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+truc
```
Comment (RobbieTheWagner) [inline test-app/app/templates/flatpickr-test.hbs:1]:
```suggestion
```

Was this supposed to be here?

File: test-app/app/templates/docs/index.md [hunk 7]
```diff
@@ -18,3 +18,19 @@ This is an Ember addon that wraps the date picker [flatpickr](http://chmln.githu
...
+
+ember-flatpickr requires `flatpickr` as a peer dependency. Install it with your favorite package manager:
+
+```bash
+npm install -D flatpickr
+pnpm add -D flatpickr
+yarn add -D flatpickr
+```
+
+### Default style
```
Comment (RobbieTheWagner) [inline test-app/app/templates/docs/index.md:30]: Should we mention themes or is that mentioned elsewhere?

Comment (AmauryD) [reply test-app/app/templates/docs/index.md:30]: There's a brief section on this page:  https://github.com/RobbieTheWagner/ember-flatpickr/blob/2b148eb050bbce62026f38dcf108ca085bcf49a5/test-app/app/templates/docs/usage.md#themes

File: test-app/app/templates/flatpickr-test.hbs (hunk 6 above)
Comment (AmauryD) [reply test-app/app/templates/flatpickr-test.hbs:1]: absolutely not, it's a french word i use for testing, it means 'thing'. I'm removing this ASAP 😅

**Review by RobbieTheWagner — APPROVED** [review]
Awesome work, thank you! There might be a few things we need to document a bit more, but I want to merge this so we don't keep getting conflicts.

**Comment by AmauryD** [comment]
> Awesome work, thank you! There might be a few things we need to document a bit more, but I want to merge this so we don't keep getting conflicts.

You're welcome ! It was a pleasure ! 😄

**Pull request merged**